src/projects/shared/api/circuitdesigner/src/public/impl/Viewport.ts

```diff
@@ -336,16 +336,16 @@ export class ViewportImpl<T extends CircuitTypes> extends MultiObservable<Viewpo
             return;
         }
 
-        const { left, right, bottom, top } = { ...{ left: 0, right: 0, bottom: 0, top: 0 }, ...this.margin };
+        const { left, right, bottom, top } = { left: 0, right: 0, bottom: 0, top: 0, ...this.margin };
 
-        const marginSize = V(left - right, bottom - top);
+        const marginSize = V(left + right, bottom + top);
         const bbox = Rect.Bounding(objs.map(({ bounds }) => bounds));
 
         const screenSize = canvasInfo.screenSize.sub(V(left, bottom));
         const worldSize = this.toWorldPos(screenSize).sub(this.toWorldPos(V(0, 0)));
 
         // Determine which bbox dimension will limit zoom level
-        const ratio = V(bbox.width / worldSize.x, bbox.height / worldSize.y);
+        const ratio = V(bbox.width / worldSize.x, -bbox.height / worldSize.y);
```

Comment:
Will the negative sign not always make it negative meaning the .x value will always be chosen in Math.max below?

Reply:
The issue was that this was always computing to be negative because worldSize.y is always negative because `this.toWorldPos(screenSize)` results in a negative value for y (and I don't really follow the math any further, I'm just copying from master and master has this same issue). I can put the `-` on the worldSize.y so it is more clear what it is negating